src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -139,7 +139,7 @@ class BlockExecution(
       if (remainingBlocks.isEmpty) {
         (executedBlocks.reverse, None)
       } else if (error.isDefined) {
-        (executedBlocks, error)
+        (executedBlocks.reverse, error)
```

Comment:
i'm struggling with this analysis in the code. Maybe as a workarroung we could add a suffix at variables name in order to know if the list is in incremental or decremental order. WDYT? 

ex.
`remainingBlocks` -> `remainingBlocksIncOrder`  // [bn=1, bn=2, bn=3]
`executedBlocks` -> `executedBlocksDecOrder`   // [bn=3, bn=2, bn=1]

And also add a comment regarding the order of the list returned by this function